lib/rubygems/request_set.rb

```diff
@@ -152,7 +152,34 @@ def install options, &block # :yields: request, installer
     @prerelease = options[:prerelease]
 
     requests = []
+    download_queue = Queue.new
 
+    # Create a thread-safe list of gems to download
+    sorted_requests.each do |req|
+      download_queue << req
+    end
+
+    # Create N threads in a pool, have them download all the gems
+    threads = Gem.configuration.concurrent_downloads.times.map do
+      # When a thread pops this item, it knows to stop running. The symbol
+      # is queued here so that there will be one symbol per thread.
+      download_queue << :stop
+
+      Thread.new do
```

Comment:
does this handle receiving `SIGINT` gracefully? (e.g. does the program exit as expected)

Reply:
I'm not sure. Added to the TODO list in the first comment.

Reply:
@segiddins sorry for the delay &mdash; forgot to follow up on this. it does in fact handle `SIGINT` gracefully.